### PR TITLE
Updating EVM tool

### LINKED ISSUE #831: Add EVM tool example
## Describe the bug

EVM tool page missing example usage.
Also link to state-test subcommand is wrong.

- [x] Missing content
- [ ] Outdated content
- [x] Wrong content
- [ ] Confusing or misleading content
- [ ] Other

## The broken page

https://besu.hyperledger.org/en/stable/HowTo/Troubleshoot/Use-EVM-Tool/#evm-tool-run-options

## Change suggestion

Add example command for running EVM tool to the run options section:

```
evm state-test <state-test> --nomemory
```

Where `<state-test>` is a specific state test, for example: https://github.com/ethereum/tests/blob/e102032ae30ef8ae0f0ec94b9cf37ac53c940cb2/GeneralStateTests/stPreCompiledContracts2/CALLCODERipemd160_5.json

Also fix the link to the state-test subcommand reference.
https://besu.hyperledger.org/en/stable/Reference/Evm-Tool/#evm-tool-state-test-options -> https://besu.hyperledger.org/en/stable/Reference/Evm-Tool/#state-test-options

### PR DESCRIPTION
**Pull request by rolandtyler**
Signed-off-by: Roland Tyler <roland.tyler@consensys.net>

## Pull request checklist

Use the following list to make sure your PR fits the Besu doc quality standard.

### Before creating the pull request

Make sure that:

- [x] [all commits in this PR are signed off for the DCO](https://wiki.hyperledger.org/display/BESU/DCO).
- [x] you read the [contribution guidelines](https://wiki.hyperledger.org/display/BESU/Contributing+to+documentation).
- [x] you have [tested your changes locally](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewTheDocumentation) before submitting them to the community for review.

### After creating your pull request and tests finished

Make sure that:

- [x] you fixed all the issues raised by the tests, if any.
- [x] you verified the rendering of your changes on [ReadTheDocs.org PR preview](https://wiki.hyperledger.org/display/BESU/MkDocs+And+Markdown+Guide#MkDocsAndMarkdownGuide-PreviewwithReadTheDocs)
  and updated the testing link (see [Testing](#testing)).

## Describe the change
Updated EVM content for clarity, minor stylistic edits.

## Issue fixed
Fixes #831 

### For content changes

- [x] Doc content
- [ ] Doc pages organisation

### For tools changes

- [ ] CircleCI workflow
- [ ] Build and QA tools (lint, vale,…)
- [ ] MkDocs templates
- [ ] MkDocs configuration
- [ ] Python dependencies
- [ ] Node dependencies and JavaScript
- [ ] ReadTheDocs configuration
- [ ] GitHub integration

## Testing
https://hyperledger-besu--863.org.readthedocs.build/en/863/

### CODE REVIEW

File: docs/HowTo/Troubleshoot/Use-EVM-Tool.md [hunk 1]
```diff
@@ -55,7 +55,25 @@ docker run -rm <docker options> hyperledger/besu-evmtool:develop <evmtool option
 The first mode of the EVM tool runs an arbitrary EVM and is invoked without an extra command.  [Command
 line options](../../Reference/Evm-Tool.md) specify the code and other contextual information.
 
-The EVM tool also has a [`state-test` sub command](../../Reference/Evm-Tool.md#evm-tool-state-test-options)
-that allows the Ethereum State Tests to be evaluated. Most of the options from EVM execution do not apply.
+The EVM tool also has a [`state-test` subcommand](../../Reference/Evm-Tool.md#state-test-options)
+that allows the [Ethereum State Tests](https://github.com/ethereum/tests/tree/e102032ae30ef8ae0f0ec94b9cf37ac53c940cb2/GeneralStateTests) to be evaluated.
```
Comment (garyschulte) [inline docs/HowTo/Troubleshoot/Use-EVM-Tool.md:59]: Do we want to link to a specific SHA within the reference tests or just link to the main branch?  e.g. https://github.com/ethereum/tests/tree/develop/GeneralStateTests 

I can see an argument for either, just want to ensure this is intentional

**Review by garyschulte — COMMENTED** [review]
Looks good.  Possibly a dead link and a question about linking specific SHAs

File: docs/HowTo/Troubleshoot/Use-EVM-Tool.md [hunk 2]
```diff
@@ -55,7 +55,25 @@ docker run -rm <docker options> hyperledger/besu-evmtool:develop <evmtool option
 The first mode of the EVM tool runs an arbitrary EVM and is invoked without an extra command.  [Command
 line options](../../Reference/Evm-Tool.md) specify the code and other contextual information.
 
-The EVM tool also has a [`state-test` sub command](../../Reference/Evm-Tool.md#evm-tool-state-test-options)
-that allows the Ethereum State Tests to be evaluated. Most of the options from EVM execution do not apply.
+The EVM tool also has a [`state-test` subcommand](../../Reference/Evm-Tool.md#state-test-options)
+that allows the [Ethereum State Tests](https://github.com/ethereum/tests/tree/develop/GeneralStateTests) to be evaluated.
```
Comment (alexandratran) [inline docs/HowTo/Troubleshoot/Use-EVM-Tool.md:59]:
```suggestion
that allows the [Ethereum state tests](https://github.com/ethereum/tests/tree/develop/GeneralStateTests) to be evaluated.
```

File: docs/Reference/Evm-Tool.md [hunk 3]
```diff
@@ -299,7 +299,7 @@ specific world state when running with `rocksdb` for `key-value-storage`.
 
 ## State test options
 
-The `state-test` sub command allows the Ethereum State Tests to be evaluated.
+The `state-test` subcommand allows the [Ethereum State Tests](https://github.com/ethereum/tests/tree/develop/GeneralStateTests) to be evaluated.
```
Comment (alexandratran) [inline docs/Reference/Evm-Tool.md:302]:
```suggestion
The `state-test` subcommand allows the [Ethereum state tests](https://github.com/ethereum/tests/tree/develop/GeneralStateTests) to be evaluated.
```

File: docs/Reference/Evm-Tool.md [hunk 4]
```diff
@@ -323,10 +323,27 @@ Set to true for EVMLab Fuzzing.
...
+    ```
+
+=== "Example"
+
+    ```bash
+    --nomemory=true
+    ```
+
+By default, when tracing operations the memory is traced for each operation.
+For memory heavy scripts, setting this option may reduce the volume of JSON output.
```
Comment (alexandratran) [inline docs/Reference/Evm-Tool.md:341]:
```suggestion
For memory heavy scripts, setting this option to `true` may reduce the volume of JSON output.
```
Can you also add backticks to the instances of "true" in the description of the previous `json` option?

**Review by alexandratran — APPROVED** [review]
A few suggestions. Looks good!

**Pull request merged**